awscli/clidocs.py

```diff
@@ -716,3 +721,48 @@ def _line_has_tag(self, line):
 
     def doc_subitems_start(self, help_command, **kwargs):
         pass
+
+
+class GlobalOptionsDocumenter:
+    """Documenter used to pre-generate global options docs."""
+
+    # Maps a condition to a string format.
+    SYNOPSIS_MAP = {
+        lambda x: x.cli_type_name != 'boolean':
+            lambda x: f"[{x.cli_name} <value>]",
+        lambda x: bool(x):
+            lambda x: f"[{x.cli_name}]"
+    }
```

Comment:
As a rule of thumb, we try to avoid using lambdas especially if it is anything complicated mainly because they are difficult to troubleshoot. If we were to go forward with this approach I'd recommend just simplifying it down to:
```python
_SYNOPSIS_TYPE_TO_FORMAT = {
     "boolean": "[{cli_name}]",
     # Add formats for other types
}
```
However, I'd recommend we do not need to generalize the logic in a map for this as we really only need to special case one type and we should just have an if statement for the boolean type.  So something like this later down in the logic:
```python
        for arg in help_command.arg_table.values():
            if arg.cli_type_name == 'boolean':
                arg_synopsis = f"[{x.cli_name}]"
            else:
                arg_synopsis = f"[{arg.cli_name} <value>]"
            help_command.doc.writeln(arg_synopsis)
```